lib/Web/Main.hs

```diff
@@ -4,18 +4,73 @@
 
 module Web.Main
   ( runServer
+  , RunServerException (..)
   ) where
 
 import Backend.Commands (runCommand)
 import Config
+import Control.Exception (Exception, throw)
+import Data.Bifunctor (Bifunctor(first))
+import Data.Either (partitionEithers)
 import Data.Proxy
 import Network.Wai.Handler.Warp (run)
+import Options.Applicative
 import Servant.Server (serve)
+import System.Environment (lookupEnv)
+import Text.Interpolation.Nyan
+import Text.Read (readEither)
 import Web.API (API)
 import Web.Server
 
+data ServerOptions = ServerOptions { serverPort :: Maybe Int }
+
+data RunServerException
+  = RunServerIncorrectPort Int
+  | RunServerNoPortSpecified
+  | RunServerEnvVarParseFail String
+  deriving stock (Eq)
+
+instance Show RunServerException where
+  show (RunServerIncorrectPort port) = "Port number should be greater than 0. Your port number is: " ++ show port
```

Comment:
*Port number should be from 0 to 65536* is a bit clearer

Reply:
Or just don't check the range at all - just pass the port number to servant and let servant do the check.